mozilla_django_oidc/middleware.py

```diff
@@ -0,0 +1,82 @@
+try:
+    from urllib import urlencode
+except ImportError:
+    from urllib.parse import urlencode
+
+import time
+
+import django
+from django.core.urlresolvers import reverse
+from django.http import HttpResponseRedirect
+from django.utils.crypto import get_random_string
+
+from mozilla_django_oidc.utils import absolutify, import_from_settings
+
+
+# Django 1.11 makes changes to how middleware work. In Django 1.11+, we want to
+# use the mixin so that our middleware works as is.
+if tuple(django.VERSION[0:2]) >= (1, 10):
+    from django.utils.deprecation import MiddlewareMixin
+else:
+    class MiddlewareMixin(object):
+        pass
+
+
+# FIXME(willkg): This doesn't appear to be used anywhere.
+def logout_url():
+    """Log out the user from Auth0."""
+    url = 'https://' + import_from_settings('OIDC_OP_DOMAIN') + '/v2/logout'
+    url += '?' + urlencode({
+        'returnTo': import_from_settings('LOGOUT_REDIRECT_URL', '/'),
+        'client_id': import_from_settings('OIDC_RP_CLIENT_ID')
+    })
+    return url
+
+
+class RefreshIDToken(MiddlewareMixin):
+    """Renews id_tokens after expiry seconds
+
+    For users authenticated with an id_token, we need to check that it's still
+    valid after a specific amount of time and if not, force them to
+    re-authenticate silently.
+
+    """
+    def process_request(self, request):
+        if ((request.method == 'GET' and
+             request.user.is_authenticated() and
+             not request.path.startswith('/oidc/') and
+             not request.is_ajax())):
+            expiration = request.session.get('oidc_id_token_expiration')
+            if expiration is not None and expiration > time.time():
```

Comment:
`time.time()` will give back seconds since epoch for that server. If all the servers in a cluster are in the same timezone, that'll work fine. If they're not then this won't work.

What's a better way to do this? Should we be using `YYYY-mm-dd HH:MM:SS` UTC timestamps and string comparisons?